sycl/include/CL/sycl/detail/common.hpp

```diff
@@ -11,9 +11,6 @@
 #include <CL/sycl/detail/defines.hpp>
 #include <CL/sycl/detail/export.hpp>
 
-// Suppress a compiler warning about undefined CL_TARGET_OPENCL_VERSION
-// Khronos ICD supports only latest OpenCL version
-#define CL_TARGET_OPENCL_VERSION 220
 #include <CL/cl.h>
```

Comment:
I'm not sure this a good idea to remove this part.

If I write a SYCL application and include sycl headers, I will get these OpenCL headers included and I think they will spawn a lot of remarks about missing `CL_TARGET_OPENCL_VERSION` definition.

Reply:
At some point it was enough to have clean build logs, but at point more `#include <CL/cl.h>` appeared in other places and now I see 11 different files including `<CL/cl.h>` and only `common.hpp` defines this macro.

You patch cleans the build logs for SYCL library build, but I think we should fix this issue for any source file including SYCL headers.

Reply:
I agree.

However, I guess users may also want to set `CL_TARGET_OPENCL_VERSION` to a specific value; or they may be using some system-provided OpenCL headers that do not use this symbol (unless it is actually mandated by the standard).

Reply:
How about
  - `sycl/detail/common.hpp` defines the symbol only if it is not already defined
  - all other headers include `sycl/detail/common.hpp` instead of `CL/cl.h`
?

Unless there is a long-term goal of *not* including `CL/cl.h` `common.hpp` ?
Then it could be further split:
  - `sycl/detail/cl.hpp` defines the symbol only if it is not already defined, and include `CL/cl.h` *etc*
  - `sycl/detail/common.hpp` and all other headers include `sycl/detail/cl.hpp` instead of `CL/cl.h`
?

Reply:
Thanks! I'm fine with both options, but it looks like splitting might be preferable as `common.hpp` has a lot of additional declarations, which doesn't seems to be quite "common".
`sycl/include/CL/cl_usm_ext.h` includes just `<CL/cl.h>` and we should probably keep it similar way.

Reply:
OK, let me try to update this PR with the proposed changes.

Reply:
Done, and rebased on the current HEAD to fix conflicts.